src/themes/material/components/chip-list.component.ts

```diff
@@ -0,0 +1,203 @@
+
+interface IChipValue {
+	index?: number;
+	value?: any;
+}
+
+interface ISelectedItem {
+	item: IChipValue;
+	action: Actions;
+}
+
+interface ICommandSet {
+	beginEdit: any;
+	endEdit: any;
+	addItem: any;
+	deleteItem: any;
+}
+
+enum Actions {
+	inactive,
+	add,
+	beginEdit,
+	endEdit,
+	delete
+}
+
+@Component({
+	selector: 'q-grid-chips',
+	templateUrl: './chip-list.tpl.html'
+})
+export class ChipListComponent implements OnInit {
+	@Input('values') chipValues: any[];
+	private shortcut = new Shortcut(new ShortcutManager());
+
+	private chipValueMap: IChipValue[];
+
+	private selectedItem: ISelectedItem = {
+		item: {},
+		action: Actions.inactive
+	};
+
+	private newItem: ISelectedItem = {
+		item: {},
+		action: Actions.inactive
+	};
+
+	get chipList(): any[] {
+		return this.chipValueMap.map((item) => item.value);
+	}
+	set chipList(values: any[]) {
+		this.chipValueMap = values.map((itemValue, itemIndex) => {
+			return <IChipValue> {
+				value: itemValue,
+				index: itemIndex
+			};
+		});
+	};
+
+	private commandSet: ICommandSet = {
+		beginEdit: new Command({
+			canExecute: () => this.selectedItem.action === Actions.beginEdit,
+			execute: () => {
+				this.selectedItem.item.value = this.chipValues[this.selectedItem.item.index];
+			},
+			shortcut: 'enter'
+		}),
+
+		endEdit: new Command({
+			canExecute: () => this.selectedItem.action === Actions.endEdit &&
+				this.selectedItem.item.value,
+			execute: () => {
+				this.chipValues[this.selectedItem.item.index] = this.selectedItem.item.value;
+
+				this.chipValueMap[this.selectedItem.item.index].value = this.selectedItem.item.value;
+				this.resetSelectedAction();
+				this.cdRef.detectChanges();
+			},
+			shortcut: 'enter'
+		}),
+
+		addItem: new Command({
+			canExecute: () => this.newItem.action === Actions.add &&
+				this.newItem.item.value && !this.findExistingItem(this.selectedItem.item.value),
+			execute: () => {
+				this.chipValues.push(this.newItem.item.value);
+
+				this.chipValueMap.push({
+					value: this.newItem.item.value,
+					index: this.chipValueMap.length
+				});
+
+				this.resetAddNewAction();
+			},
+			shortcut: 'enter'
+		}),
+
+		deleteItem: new Command({
+			canExecute: () => this.selectedItem.action === Actions.delete,
+			execute: () => {
+				this.chipValues.splice(this.selectedItem.item.index, 1);
+
+				this.chipValueMap = this.chipValues.map((itemValue, itemIndex) => {
+					return <IChipValue> {
+						value: itemValue,
+						index: itemIndex
+					};
+				});
+				this.resetSelectedAction();
+				this.cdRef.detectChanges();
+			}
+		})
+	};
+
+	constructor(private cdRef: ChangeDetectorRef) {
+		const manager = new CommandManager();
+
+		const shortcutCommands = [this.commandSet.endEdit, this.commandSet.addItem];
+		const otherCommands = [this.commandSet.endEdit, this.commandSet.addItem, this.commandSet.deleteItem];
+
+		this.shortcut.register(manager, shortcutCommands);
```

Comment:
does it matter if command has shortcut or not? can we not split commands to this groups?